lib/actions-oclif/devices/supported.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright 2016-2019 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Command, flags } from '@oclif/command';
+import * as SDK from 'balena-sdk';
+import { stripIndent } from 'common-tags';
+import * as _ from 'lodash';
+
+import * as cf from '../../utils/common-flags';
+import { CommandHelp } from '../../utils/oclif-utils';
+
+interface FlagsDef {
+	discontinued: boolean;
+	help: void;
+	json?: boolean;
+	verbose?: boolean;
+}
+
+interface DeviceTypeWithAliases extends SDK.DeviceType {
+	aliases?: string[];
+}
+
+export default class DevicesSupportedCmd extends Command {
+	public static description = stripIndent`
+		List the supported device types (like 'raspberrypi3' or 'intel-nuc').
+
+		List the supported device types (like 'raspberrypi3' or 'intel-nuc').
```

Comment:
Is this a duplicate line?

Reply:
It is an intentional duplicate because of they way the oclif actions process / parse this string for help output. The first line is used in at the very top of help output, and the duplicate line starts the DESCRIPTION block:
```
$ ./bin/balena help devices supported
List the supported device types (like 'raspberrypi3' or 'intel-nuc').

USAGE
  $ balena devices supported

OPTIONS
  ...

DESCRIPTION
  List the supported device types (like 'raspberrypi3' or 'intel-nuc').

  The --verbose option adds extra columns/fields to the output, including the
  "STATE" column whose values are one of 'beta', 'release' or 'discontinued'.
  ...
```
The first line is also the one used in `balena help -v`.

Reply:
Ah, good to know, thanks.